examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -0,0 +1,683 @@
+              new SizedBox(),
+              new Padding(
+                padding: const EdgeInsets.only(top: 24.0, bottom: 4.0),
+                child: new Text('Steps', style: headingStyle)
+              ),
+            ])
+          )..addAll(config.recipe.steps.map(
+            (RecipeStep step) {
+              return _buildItemRow(step.duration ?? '', step.description);
+            }
+          ))
+        )
+      )
+    );
+
+    // The full page content with the recipe's image behind it. This
+    // adjusts based on the size of the screen. If the recipe sheet touches
+    // the edge of the screen, use a slightly different layout.
+    bool isFavorite = favoriteRecipes.contains(config.recipe);
+    Size screenSize = MediaQuery.of(context).size;
+    bool fullWidth = (screenSize.width < _kRecipePageMaxWidth);
+    const double fabHalfSize = 28.0;  // TODO(mpcomplete): needs to adapt to screen size
+    Widget container = new DecoratedBox(
+      decoration: new BoxDecoration(
+        backgroundImage: new BackgroundImage(
+          image: DefaultAssetBundle.of(context).loadImage(config.recipe.imagePath),
+          alignment: FractionalOffset.topCenter,
+          fit: fullWidth ? ImageFit.fitWidth : ImageFit.cover
+        )
+      ),
+      child: new Align(
+        alignment: FractionalOffset.bottomCenter,
+        child: new Block(children: <Widget>[
+          new Padding(
+            padding: new EdgeInsets.only(top: screenSize.height*0.3),
+            child: new Stack(children: <Widget>[
+              new Padding(
+                padding: new EdgeInsets.only(top: fabHalfSize),
+                child: new SizedBox(
+                  width: fullWidth ? null : _kRecipePageMaxWidth,
+                  child: recipeWidget
+                )
+              ),
+              new Positioned(
+                right: 16.0,
+                child: new FloatingActionButton(
+                  child: new Icon(icon: isFavorite ? Icons.favorite : Icons.favorite_border),
+                  onPressed: _toggleFavorite
+                )
+              )
+            ])
+          )
+        ])
+      )
+    );
+
+    PopupMenuItem<String> _menuItem(IconData icon, String label) {
+      return new PopupMenuItem<String>(
+        value: 'a',
+        child: new Row(children: <Widget>[
+          new Padding(
+            padding: const EdgeInsets.only(right: 24.0),
+            child: new Icon(icon: icon, color: Colors.black54)
+          ),
+          new Text(label, style: descriptionStyle)
+        ])
+      );
+    }
+
+    // Overlays a transparent app bar onto the page.
+    final double statusBarHeight = MediaQuery.of(context).padding.top;
+    return new Padding(
+      padding: new EdgeInsets.only(top: statusBarHeight),
+      child: new Stack(children: <Widget>[
+        container,
+        new AppBar(
+          backgroundColor: Colors.transparent,
+          leading: new IconButton(
```

Comment:
Dunno, just matching the spec and other demos. Normally the Scaffold adds the back button to its AppBar, but this one is not part of a Scaffold. Scaffold doesn't let the AppBar cover the content (AFAICT).